src/main/scala/com/databricks/spark/sql/perf/Benchmark.scala

```diff
@@ -66,25 +66,6 @@ abstract class Benchmark(
     defaultParallelism = sparkContext.defaultParallelism,
     buildInfo = buildInfo)
 
-  /**
-   * A Variation represents a setting (e.g. the number of shuffle partitions or if tables
-   * are cached in memory) that we want to change in a experiment run.
-   * A Variation has three parts, `name`, `options`, and `setup`.
-   * The `name` is the identifier of a Variation. `options` is a Seq of options that
-   * will be used for a query. Basically, a query will be executed with every option
-   * defined in the list of `options`. `setup` defines the needed action for every
-   * option. For example, the following Variation is used to change the number of shuffle
-   * partitions of a query. The name of the Variation is "shufflePartitions". There are
-   * two options, 200 and 2000. The setup is used to set the value of property
-   * "spark.sql.shuffle.partitions".
-   *
-   * {{{
-   *   Variation("shufflePartitions", Seq("200", "2000")) {
-   *     case num => sqlContext.setConf("spark.sql.shuffle.partitions", num)
-   *   }
-   * }}}
-   */
-  case class Variation[T](name: String, options: Seq[T])(val setup: T => Unit)
```

Comment:
No, this is just moving the classes out, otherwise the scala compiler gets
very confused by the path-dependent types when there is an issue elsewhere.

On Tue, Jun 21, 2016 at 12:21 PM, Yin Huai <notifications@github.com> wrote:

> In src/main/scala/com/databricks/spark/sql/perf/Benchmark.scala
> <https://github.com/databricks/spark-sql-perf/pull/69#discussion_r67934524>
> :
>
> > -   * A Variation has three parts, `name`, `options`, and `setup`.
> > -   * The `name` is the identifier of a Variation. `options` is a Seq of options that
> > -   * will be used for a query. Basically, a query will be executed with every option
> > -   * defined in the list of `options`. `setup` defines the needed action for every
> > -   * option. For example, the following Variation is used to change the number of shuffle
> > -   * partitions of a query. The name of the Variation is "shufflePartitions". There are
> > -   * two options, 200 and 2000. The setup is used to set the value of property
> > -   * "spark.sql.shuffle.partitions".
> > -   *
> > -   * {{{
> > -   *   Variation("shufflePartitions", Seq("200", "2000")) {
> > -   *     case num => sqlContext.setConf("spark.sql.shuffle.partitions", num)
> > -   *   }
> > -   * }}}
> > -   */
> > -  case class Variation[T](name: String, options: Seq[T])(val setup: T => Unit)
>
> Are we making any actual changes? Or, we are just moving these classes?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/databricks/spark-sql-perf/pull/69/files/5ce08a6f220bd478f8c946a9b1b680affe495389#r67934524>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHPjAZTwptJx-RpZdclokmzNisGViYoaks5qODmvgaJpZM4I6PAR>
> .
>